WorkloadManagementSystem/Agent/SiteDirector.py

```diff
@@ -341,67 +354,208 @@ def getQueues( self, resourceDict ):
 
         It basically just calls self.submitJobs() method
     """
 
     if not self.queueDict:
-      self.log.warn( 'No site defined, exiting the cycle' )
+      self.log.warn('No site defined, exiting the cycle')
       return S_OK()
 
+    # get list of usable sites within this cycle
+    result = self.siteClient.getUsableSites()
+    if not result['OK']:
+      return result
+    self.siteMaskList = result.get('Value', [])
+
+    if self.rssFlag:
+      ceNamesList = [queue['CEName'] for queue in self.queueDict.itervalues()]
+      result = self.rssClient.getElementStatus(ceNamesList, "ComputingElement")
+      if not result['OK']:
+        self.log.error("Can not get the status of computing elements",
+                       " %s" % result['Message'])
+      self.ceMaskList = [ceName for ceName in result['Value'].iterkeys() if result['Value'][ceName]
+                         ['all'] in ('Active', 'Degraded')]
+
+    self.matcherClient = MatcherClient()
     result = self.submitJobs()
     if not result['OK']:
-      self.log.error( 'Errors in the job submission: ', result['Message'] )
+      self.log.error('Errors in the job submission: ', result['Message'])
 
-    cyclesDone = self.am_getModuleParam( 'cyclesDone' )
+    cyclesDone = self.am_getModuleParam('cyclesDone')
     if self.updateStatus and cyclesDone % self.pilotStatusUpdateCycleFactor == 0:
       result = self.updatePilotStatus()
       if not result['OK']:
-        self.log.error( 'Errors in updating pilot status: ', result['Message'] )
+        self.log.error('Errors in updating pilot status: ', result['Message'])
+
+    return S_OK()
+
+  def submitJobs(self):
+    """ Go through defined computing elements and submit jobs if necessary and possible
+
+        :return: S_OK/S_ERROR
+    """
+
+    # First, we check if we want to submit pilots at all, and also where
+    submit, anySite, jobSites, testSites = self._ifAndWhereToSubmit()
+    if not submit:
+      self.log.notice("Not submitting any pilots at this cycle")
+      return S_OK()
+
+    # From here on we assume we are going to (try to) submit some pilots
+    self.log.debug("Going to try to submit some pilots")
+
+    queues = self.queueDict.keys()
+    random.shuffle(queues)
+    self.log.verbose("Queues treated: %s" % ','.join(queues))
+
+    self.totalSubmittedPilots = 0
+
+    for queue in queues:
+      # now submitting to the single queues
+      self.log.verbose("Evaluating queue %s" % queue)
+
+      # are we going to submit pilots to this specific queue?
+      if not self._allowedToSubmit(queue, anySite, jobSites, testSites):
+        continue
+
+      if 'CPUTime' in self.queueDict[queue]['ParametersDict']:
+        queueCPUTime = int(self.queueDict[queue]['ParametersDict']['CPUTime'])
+      else:
+        self.log.warn('CPU time limit is not specified for queue %s, skipping...' % queue)
+        continue
+      if queueCPUTime > self.maxQueueLength:
```

Comment:
I might be a bit naive or optimist here, but isn't this a good opportunity to assign correct names 'work' related rather than time where needed ?

Reply:
I don't to change it only here now. Once we'll make a proper grep and change these names, not now.